### ISSUE TITLE
Pipeline

### ISSUE THREAD

**Issue by ShJacub** [LukasHedegaard/ride#19]
Hello! Thank for your contribution in computer science! Is there any way to adjust learning pipeline or this tool is suitable only for classification. For example, If I want to use it for object detection what should I do? Thank you for your attention.

**Comment by LukasHedegaard** [comment]
Hi @ShJacub . The tool can be used for many different ML tasks, including object detection. 

To integrate your own object detection dataset, your should subclass `[RideDataset](https://github.com/LukasHedegaard/ride/blob/f6f9aadd4070fd72d874b8e934c7e44f92c02d08/ride/core.py#L236)` and return the appropriate loaders. You then "mix-in" your dataloader class by inheriting from it in your main module (the one that derives from RideModule).

You can either use the "--loss" argument to select one of the pre-configured losses (multilabel_margin_loss,smooth_l1_loss,kl_div,nll_loss,l1_loss,multi_margin_loss,poisson_nll_loss,soft_margin_loss,hinge_embedding_loss,binary_cross_entropy_with_logits,cross_entropy,multilabel_soft_margin_loss,mse_loss,binary_cross_entropy) or provide a custom loss by simply defining a loss function in your main class (the one that derives from RideModule).

You can provide custom metrics by subclassing the `MetricMixin`, e.g. as done for the [TopKAccuracy](https://github.com/LukasHedegaard/ride/blob/f6f9aadd4070fd72d874b8e934c7e44f92c02d08/ride/metrics.py#L213) and by inheriting from it in your main module  (the one that derives from RideModule).

All in all your class definition would be something like:

```python
class MyObjectDetector(
    ride.RideModule,
    MyCustomMetricWhichInheritedFromMetrixMixin,
    ride.optimizers.SgdOneCycleOptimizer, # for example
    MyCustomDatasetWhichInheritedFromRideDataset,
):
    def __init__(self, hparams):
      ...
```

I hope this helps. I may add a more complete example for object detection in the documentation at some point.
